components/inputs/input-time.js

```diff
@@ -0,0 +1,156 @@
+	if (match !== null) {
+		if (match.length > 1) {
+			hour = parseInt(match[1]);
+			if (isNaN(hour) || hour < 0 || hour > 23) {
+				hour = 0;
+			}
+		}
+		if (match.length > 2) {
+			minute = parseInt(match[2]);
+			if (isNaN(minute) || minute < 0 || minute > 59) {
+				minute = 0;
+			}
+		}
+		if (match.length > 3) {
+			second = parseInt(match[3]);
+			if (isNaN(second) || second < 0 || second > 59) {
+				second = 0;
+			}
+		}
+	}
+	const time = new Date(TODAY.getFullYear(), TODAY.getMonth(), TODAY.getDate(), hour, minute, second);
+	return time;
+}
+
+class InputTime extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			label: { type: String },
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			value: { type: String },
+			_formattedValue: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ inputStyles, inputLabelStyles,
+			css`
+				:host {
+					display: inline-block;
+					width: 100%;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				label {
+					display: block;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.labelHidden = false;
+		this._value = formatValue(DEFAULT_VALUE);
+		this._formattedValue = formatTime(DEFAULT_VALUE);
+	}
+
+	get value() { return this._value; }
+	set value(val) {
+		const oldValue = this.value;
+		const time = parseValue(val);
+		this._value = formatValue(time);
+		this._formattedValue = formatTime(time);
+		this.requestUpdate('value', oldValue);
+	}
+
+	render() {
+		const input = html`
+			<input
+				aria-label="${ifDefined(this._getAriaLabel())}"
+				@change="${this._handleChange}"
+				class="d2l-input"
+				?disabled="${this.disabled}"
+				@keypress="${this._handleKeypress}"
+				.value="${this._formattedValue}">
+		`;
+		if (this.label && !this.labelHidden) {
+			return html`
+				<label>
+					<span class="d2l-input-label">${this.label}</span>
+					${input}
+				</label>`;
+		}
+		return input;
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('.d2l-input');
+		if (elem) elem.focus();
+	}
+
+	getTime() {
+		const time = parseValue(this.value);
+		return {
+			hour: time.getHours(),
+			minute: time.getMinutes(),
+			second: time.getSeconds()
+		};
+	}
+
+	_getAriaLabel() {
+		if (this.label && this.labelHidden) {
+			return this.label;
+		}
+		if (this.hasAttribute('aria-label')) {
+			return this.getAttribute('aria-label');
+		}
+		return undefined;
+	}
+
+	async _handleChange(e) {
+		const value = e.target.value;
+		const time = parseTime(value);
+		this._formattedValue = value;
+		await this.updateComplete;
+		if (time === null) {
+			this._formattedValue = formatTime(parseValue(this.value));
+		} else {
+			this.value = formatValue(time);
+			// Change events aren't composed, so we need to re-dispatch
+			this.dispatchEvent(new CustomEvent(
+				'd2l-time-input-changed',
+				{bubbles: true, composed: false}
```

Comment:
Is this meant to be composed?

Reply:
The text and search inputs both don't use composed events... guess it depends whether we want these crossing shadow DOM boundaries. Not sure if that decision was arbitrary or not, but I think we should be consistent at least, unless there's a good reason to deviate.

Reply:
If I remember correctly, we eventually plan on having a full `date-time` component that contains the individual `date` and `time` input components, right? Is there any reason we would want events from the sub-components to be read at a higher level there, or are we just lumping both into a changed event for the whole thing? 🤔

Reply:
I think we actually would want those events to specifically NOT bubble up -- the date-time component would want them trapped and it would send its own `change` event that contains the combined date and time.

Reply:
i.e. I'm leaning towards leaving this as not composed.

Reply:
Cool - I wasn't sure what to think of it, based on the comment which made me think we meant to make it composed.

Reply:
That's a good point though -- Scy can you remove the comment as it was intended for text-inputs which do just need to re-send the original event. In our case, we trap the text-input event and send our own intentionally.